### ISSUE TITLE
Files with certain modification timestamps are not properly transmitted while using the "serve ftp" feature

### ISSUE THREAD

**Issue by ghost** [rclone/rclone#6785]
#### Output of `rclone version`

rclone v1.61.1
- os/version: debian 11.6 (64 bit)
- os/kernel: 5.10.0-20-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.19.4
- go/linking: static
- go/tags: none

#### Describe the issue

I'm having difficulty explaining this issue, so please bear with me.
While testing Rclone's "serve ftp" feature along with the corresponding remote, I noticed that some files were being ignored entirely. I eventually discovered that it has something to do with their last modification time.
It appears that Rclone's FTP remote disregards these files entirely, whereas other FTP clients display them with their name and modification time intermingled.

Creating a file using this Touch syntax and placing it in a directory served over FTP by Rclone would demonstrate this behavior:
```
touch -t $(date -d '@1456772182' +'%Y%m%d%H%M.%S') BrokenFile.txt
```

**Comment by animosity22** [comment]
Without sharing some details/config file, log, it's difficult to understand what you are trying to report.

We have the template for a reason and not sure why you've ignored it.

**Comment by ncw** [comment]
I think this is a bug in the server.

What is happening is that it lists the files like this

```
2023/02/24 15:24:04 DEBUG : FTP Rx: "150 Opening ASCII mode data connection for file list"
2023/02/24 15:24:04 DEBUG : FTP Rx: "-rw-rw-r-- 1 ncw ncw            0 Feb 29 18:56 BrokenFile.txt"
2023/02/24 15:24:04 DEBUG : FTP Rx: "-rw-rw-r-- 1 ncw ncw            0 Feb 24 15:23 OKfile.txt"
2023/02/24 15:24:04 DEBUG : FTP Rx: "226 Closing data connection, sent 122 bytes"
```

But the files have these dates

```
-rw-rw-r-- 1 ncw ncw 0 Feb 29  2016 BrokenFile.txt
-rw-rw-r-- 1 ncw ncw 0 Feb 24 15:23 OKfile.txt
```

```
-rw-rw-r-- 1 ncw ncw 0 2016-02-29 18:56:22.000000000 +0000 BrokenFile.txt
-rw-rw-r-- 1 ncw ncw 0 2023-02-24 15:23:44.402103204 +0000 OKfile.txt
```

FTP servers which don't support MLST are supposed to report the dates as either `Feb 24 15:23` if within the last year or `Feb 29  2016` if not.

Rclone's FTP server is reporting `Feb 29 18:56` for the file older than 2016 which is being interpreted as `Feb 29 18:56 2023` which of course does not exist.

The [FTP library rclone is using](https://github.com/jlaffaye/ftp) is silently dropping this file.

The bug appears to be in the [FTP library rclone is using to serve](https://gitea.com/goftp/server) the files.

It looks also like this bug has been fixed already: https://gitea.com/goftp/server/commit/d07a820aac35e68784786d8ff5dd37144f274c4a

Rclone is using goftp.io/server v0.4.1 so it is easy to update to master

Can you try this please?

[v1.62.0-beta.6741.98d8fa455.fix-6785-ftp-server-times](https://beta.rclone.org/branch/fix-6785-ftp-server-times/v1.62.0-beta.6741.98d8fa455.fix-6785-ftp-server-times/) on branch [fix-6785-ftp-server-times](https://github.com/rclone/rclone/tree/fix-6785-ftp-server-times) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.62
